virtio-devices/src/block_io_uring.rs

```diff
@@ -331,15 +330,20 @@ impl BlockIoUring {
     #[allow(clippy::too_many_arguments)]
     pub fn new(
         id: String,
-        mut disk_image: File,
+        disk_image: Box<dyn DiskFile>,
         disk_path: PathBuf,
         is_disk_read_only: bool,
         iommu: bool,
         num_queues: usize,
         queue_size: u16,
         seccomp_action: SeccompAction,
     ) -> io::Result<Self> {
-        let disk_size = disk_image.seek(SeekFrom::End(0))? as u64;
+        let disk_size = disk_image.size().map_err(|e| {
```

Comment:
Seeking to the end is the reliable way to do this.

Reply:
Yes that's why I moved back to the old code :)